### PR TITLE
Update apkid rules mac linux

### PR DESCRIPTION
**Pull request by superpoussin22**
### Describe the Pull Request

```
Update APKiD rule sat setup stage or in standalone 
Will provide more regular update of rules 
```

### Checklist for PR

- [X] Run MobSF unit tests (http://your-mobsf-@ip:8000/tests/ or python3 manage.py test).
- [X] Tested Working on Linux, Mac,
- [X] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)
- [X] Coding style (indentation, etc)

### Additional Comments (if any)

```
Windows will follow
```

### CODE REVIEW

File: scripts/update_apkid.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+script_path=$(dirname $0)
+if [ "$script_path" != "scripts" ] && [ "$script_path" != "./scripts" ]; then
+    echo 'Please run script from MobSF directory '
+    echo './scripts/update_apkid.sh '
+    exit  1
+fi
+current_dir=$(pwd)
+rules_dir=${current_dir}/venv/lib/python3.6/site-packages/apkid/rules/
```
Comment (ajinabraham) [inline scripts/update_apkid.sh]: What if this directory is different for user?

File: scripts/update_apkid.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+unamestr=`uname`
+if [[ "$unamestr" == 'Darwin' ]]; then
+  export ARCHFLAGS="-arch x86_64"
+  export LDFLAGS="-L/usr/local/opt/openssl/lib"
+  export CFLAGS="-I/usr/local/opt/openssl/include"
+fi
+git clone https://github.com/rednaga/APKiD.git 
+cd APKiD 
+python3 prep-release.py 
+cp apkid/rules/rules.yarc ../venv/lib/python3.6/site-packages/apkid/rules/
```
Comment (ajinabraham) [inline scripts/update_apkid.sh]: We don't control this `/venv/lib/python3.6/`

We might want to keep it under `MalwareAnalyzer/tools/`

File: setup.sh [hunk 3]
```diff
@@ -18,6 +18,8 @@ if [[ "$unamestr" == 'Darwin' ]]; then
 fi
 echo '[INSTALL] Installing Requirements'
 pip install -r requirements.txt
+echo '[INSTALL] Updating APKiD rules'
+bash scripts/update_apkid.sh
```
Comment (ajinabraham) [inline setup.sh]: My concern is this script should be fail safe. if it failed is should not affect the rest of the setup process.

File: scripts/update_apkid.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    exit  1
+fi
+current_dir=$(pwd)
+rules_dir=${current_dir}/venv/lib/python3.6/site-packages/apkid/rules/
+virtualenv venv -p python3
+source venv/bin/activate
+platform='unknown'
+unamestr=`uname`
+if [[ "$unamestr" == 'Darwin' ]]; then
+  export ARCHFLAGS="-arch x86_64"
```
Comment (ajinabraham) [inline scripts/update_apkid.sh]: Is openssl required for generating APKiD rules ?

File: scripts/update_apkid.sh (hunk 1 above)
Comment (superpoussin22) [reply scripts/update_apkid.sh]: we need an absolute path in python script so I need to calculate the path

File: scripts/update_apkid.sh (hunk 2 above)
Comment (superpoussin22) [reply scripts/update_apkid.sh]: will do :)

File: scripts/update_apkid.sh (hunk 4 above)
Comment (superpoussin22) [reply scripts/update_apkid.sh]: will clean that

File: scripts/update_apkid.sh (hunk 2 above)
Comment (superpoussin22) [reply scripts/update_apkid.sh]: I use now variable to calculate corrdctly the path so if tomorrow we use Python 3.7 it shouls work :)

File: setup.sh (hunk 3 above)
Comment (superpoussin22) [reply setup.sh]: I'm on it

File: scripts/update_apkid.sh (hunk 4 above)
Comment (superpoussin22) [reply scripts/update_apkid.sh]: removed

File: scripts/update_apkid.sh (hunk 1 above)
Comment (superpoussin22) [reply scripts/update_apkid.sh]: use a better way to calculate path

File: setup.sh (hunk 3 above)
Comment (superpoussin22) [reply setup.sh]: we can also remove it from install process so user launch it if they want :)

Comment (superpoussin22) [reply setup.sh]: but with last change it shouldn't fail

File: scripts/update_apkid.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
+script_path=$(dirname $0)
+if [ "$script_path" != "scripts" ] && [ "$script_path" != "./scripts" ]; then
+    echo 'Please run script from MobSF directory '
+    echo './scripts/update_apkid.sh '
+    exit  1
+fi
+virtualenv venv -p python3
```
Comment (ajinabraham) [inline scripts/update_apkid.sh]: virtualenv is already created and sourced by setup.sh

File: scripts/update_apkid.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+source venv/bin/activate
+virtual_env=$(echo $VIRTUAL_ENV)
+if [ -e "$virtual_env" ]; then
+    apkid_dir="${virtual_env}/lib/python3.6/site-packages/apkid"
+    if [ ! -d "$apkid_dir" ];then
+      pip install apkid
```
Comment (ajinabraham) [inline scripts/update_apkid.sh]: The rednaga fork of yara-python needs to be installed first right? If we are having script for updating APKiD rules, then why not one for installing yara-python ?

File: scripts/update_apkid.sh [hunk 7]
```diff
@@ -0,0 +1,30 @@
+script_path=$(dirname $0)
+if [ "$script_path" != "scripts" ] && [ "$script_path" != "./scripts" ]; then
+    echo 'Please run script from MobSF directory '
+    echo './scripts/update_apkid.sh '
+    exit  1
+fi
+if [ ! -d venv/bin/ ]; then
+   virtualenv venv -p python3
+   source venv/bin/activate
```
Comment (ajinabraham) [inline scripts/update_apkid.sh]: Is this source and venv check required? We already sourced from the parent script.
https://github.com/MobSF/Mobile-Security-Framework-MobSF/blob/88646375b1180dd5ff9aed4fb6dd417e91ce4ac8/setup.sh#L11

**Comment by ajinabraham** [comment]
I will test this out. Need some time to test in my windows box. 
I haven't just settled yet.

**Comment by ajinabraham** [comment]
I am going to remove apkid till it's fixed in upstream.

**Pull request merged**